middlewares/compress_test.go

```diff
@@ -0,0 +1,70 @@
+package middlewares
+
+import (
+	"bytes"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/NYTimes/gziphandler"
+	"github.com/containous/traefik/testhelpers"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	acceptEncodingHeader = "Accept-Encoding"
+	varyHeader           = "Vary"
+	gzip                 = "gzip"
+)
+
+func TestShouldCompressWhenNoContentEncodingHeader(t *testing.T) {
+	handler := &Compress{}
+
+	req := testhelpers.MustNewRequest(http.MethodGet, "http://localhost", nil)
+	req.Header.Add(acceptEncodingHeader, gzip)
+
+	baseBody := generateBytes(gziphandler.DefaultMinSize)
+	next := func(rw http.ResponseWriter, r *http.Request) {
+		rw.Write(baseBody)
+	}
+	rw := httptest.NewRecorder()
+
+	handler.ServeHTTP(rw, req, next)
+
+	assert.Equal(t, gzip, rw.Header().Get(contentEncodingHeader))
+	assert.Equal(t, acceptEncodingHeader, rw.Header().Get(varyHeader))
+
+	if bytes.Compare(rw.Body.Bytes(), baseBody) == 0 {
+		t.Errorf("got %v,\n expected a compressed body.", rw.Body.Bytes())
+	}
+}
+
+func TestShouldNotCompressWhenContentEncodingHeader(t *testing.T) {
+	handler := &Compress{}
+
+	req := testhelpers.MustNewRequest(http.MethodGet, "http://localhost", nil)
+	req.Header.Add(acceptEncodingHeader, gzip)
+	req.Header.Add(contentEncodingHeader, gzip)
+
+	baseBody := generateBytes(gziphandler.DefaultMinSize)
+
+	next := func(rw http.ResponseWriter, r *http.Request) {
+		rw.Write(baseBody)
+	}
+
+	rw := httptest.NewRecorder()
+	handler.ServeHTTP(rw, req, next)
+
+	assert.Equal(t, "", rw.Header().Get(contentEncodingHeader))
+	assert.Equal(t, "", rw.Header().Get(varyHeader))
+
+	assert.EqualValues(t, rw.Body.Bytes(), baseBody)
+}
```

Comment:
Hmm sorry, I find most of these arguments too general and abstract.

The flexibility point sounds like a bet into the future that only some parts of the tests will need to be adjusted. It could just as well be that a part common to both tests will need to be changed, so separate tests could be more costly. We don't know what part might change; we do know though that we can save lines and improve readability by taking a table-driven approach _now_, possibly changing again in the future.

So I don't agree but I neither bother too much as it's just about two tests. :-)